packages/ag-charts-community/src/chart/chart.ts

```diff
@@ -1047,26 +1048,46 @@ export abstract class Chart extends Observable {
     protected animationRect?: BBox;
 
     // x/y are local canvas coordinates in CSS pixels, not actual pixels
-    private pickSeriesNode(point: Point, exactMatchOnly: boolean, maxDistance?: number): PickedNode | undefined {
+    private pickNode(
+        point: Point,
+        useSeriesTooltipRange: boolean,
+        exactMatchOnly: boolean,
+        maxDistance?: number
+    ): PickedNode | undefined {
         const start = performance.now();
 
         // Disable 'nearest match' options if looking for exact matches only
-        const pickModes = exactMatchOnly ? [SeriesNodePickMode.EXACT_SHAPE_MATCH] : undefined;
+        const sharedPickModes =
+            !useSeriesTooltipRange && exactMatchOnly ? [SeriesNodePickMode.EXACT_SHAPE_MATCH] : undefined;
 
         // Iterate through series in reverse, as later declared series appears on top of earlier
         // declared series.
-        const reverseSeries = [...this.series].reverse();
+        const reverseSeries: Series<SeriesNodeDatum, SeriesProperties<object[]>>[] = [...this.series].reverse();
 
         let result: { series: Series<any, any>; datum: SeriesNodeDatum; distance: number } | undefined;
         for (const series of reverseSeries) {
             if (!series.visible || !series.rootGroup.visible) {
                 continue;
             }
+            const tooltipRange = series.properties.tooltip.range;
+            let pickModes: SeriesNodePickMode[] | undefined;
+            if (useSeriesTooltipRange) {
+                if (tooltipRange === 'exact') {
+                    pickModes = [SeriesNodePickMode.EXACT_SHAPE_MATCH];
+                } else {
+                    pickModes = undefined;
+                }
+            } else {
+                pickModes = sharedPickModes;
+            }
+
```

Comment:
Nit: Just wondering, since we have distinct entry points for different values of `useSeriesTooltipRange`, whether we could move some of this logic here up to the callers of this function, rather than passing the boolean in?

Reply:
Good idea. I'll try that.